src/components/SoraCard/SoraCardIntroPage.vue

```diff
@@ -71,18 +114,39 @@ export default class SoraCardIntroPage extends Mixins(mixins.LoadingMixin, Trans
     return '';
   }
 
+  getBtnLoading(): boolean {
+    return this.loading || !this.isPriceCalculated;
+  }
+
+  openX1(): void {
+    this.showX1Dialog = true;
+  }
+
+  bridgeTokens(): void {
+    if (!this.isEuroBalanceEnough) {
+      router.push({ name: PageNames.Bridge, params: { xorToDeposit: this.xorToDeposit.toString() } });
+    }
+  }
+
+  issueCardByPaywings(): void {
+    this.showPaywingsDialog = true;
+  }
+
   handleConfirm(): void {
     if (!this.isLoggedIn) {
       router.push({ name: PageNames.Wallet });
       return;
     }
-    if (!this.isEuroBalanceEnough) {
-      router.push({ name: PageNames.Bridge, params: { xorToDeposit: this.xorToDeposit.toString() } });
-    }
 
     this.$emit('confirm-apply');
   }
 
+  loginUser(): void {
+    clearTokensFromSessionStorage();
+    const userApplied = true;
+    this.$emit('confirm-apply', userApplied);
+  }
+
   async priceLoading(): Promise<void> {
     this.isPriceCalculated = false;
     await delay(800);
```

Comment:
at this period of time (approximately) there is request to get EURO/XOR ration and calculate the amount to deposit for a particular user

Reply:
previously, in the design, we had to show this amount on the button directly and because of this calculations there was changes that would be better hidden from user. For now decided not to remove this functionality to stop user from interacting with buttons before everything is set (all calculations and rendering is done)